client/state/routing/middleware.js

```diff
@@ -13,6 +13,11 @@ import { ROUTE_SET } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
 const LAST_PATH = 'last_path';
+const ALLOWED_PATHS = /^\/(stats|plans|view|posts|pages|media|types|themes|sharing|people|plugins|domains)/i;
```

Comment:
I'd eventually like to be more specific here (i.e. only save `/stats/:slug?`, or `/posts/:slug?` for example & not anything more drilled down than that), but we can add that in a separate PR after we think on it a bit :)